File: OutOfOffice.Contracts/Models/ApprovalRequest.cs
namespace OutOfOffice.Contracts.Models;

public class ApprovalRequest
{
    public int Id { get; set; }
    public int ApproverId { get; set; }
    public int LeaveRequestId { get; set; }
    public int StatusId { get; set; }
    public string Comment { get; set; }

    public Employee Approver { get; set; }
    public LeaveRequest LeaveRequest { get; set; }
    public ApprovalRequestStatus Status { get; set; }
}

File: outofoffice.client/src/app/models/approvalRequest.ts
import { Employee } from './employee'
import { LeaveRequest } from './leaveRequest'

export interface ApprovalRequest {
  id: number
  comment: string | null
  approver: Employee
  leaveRequest: LeaveRequest
  status: ApprovalRequestStatus
}

export interface ApprovalRequestStatus {
  id: number
  name: string
}

export interface ApprovalRequestPayload {
  comment: string | null
  statusId: number
}